src/routes/(app)/product/[id]/+page.svelte

```diff
@@ -103,6 +104,25 @@
 		return true;
 	}
 	const { t, locale } = useI18n();
+	const jsonLd = {
+		'@context': 'https://schema.org/',
+		'@type': 'Product',
+		name: data.product.name,
+		image: `${$page.url.origin}/picture/raw/${data.pictures[0]._id}/format/${data.pictures[0].storage.formats[0].width}`,
+		description: data.product.description,
+		offers: {
+			'@type': 'Offer',
+			price: data.product.price.amount,
+			priceCurrency: data.product.price.currency
+		}
+	};
+
+	if (data.product.actionSettings.googleShopping.visible) {
+		const script = document.createElement('script');
+		script.type = 'application/ld+json';
+		script.textContent = JSON.stringify(jsonLd);
+		document.head.appendChild(script);
+	}
```

Comment:
I don't think you can call this server-side

Can you not create the `script` tag directly in the page content? (or in `svelte:head`)

See also this article: https://navillus.dev/blog/json-ld-in-sveltekit

Reply:
If i create new script i will not be able to access `data`.

Reply:
Maybe i can use microdonnée in page content

 ```
<div itemtype="https://schema.org/Product" itemscope>
      <meta itemprop="mpn" content="925872" />
      <meta itemprop="name" content="Executive Anvil" />
      <link itemprop="image" href="https://example.com/photos/16x9/photo.jpg" />
      <link itemprop="image" href="https://example.com/photos/4x3/photo.jpg" />
      <link itemprop="image" href="https://example.com/photos/1x1/photo.jpg" />
```

Reply:
look at the article, at the end

Reply:
to clarify:

- can you script to add the script tag in `svelte:head` ?
- **if** it doesn't work, you can use `@html` in `svelte:head` (like in the article) to render the LD+JSON script tag